docs/csharp/programming-guide/concepts/attributes/index.md

```diff
@@ -1,171 +1,124 @@
-  
--   Describing the security requirements for methods.  
-  
--   Specifying characteristics used to enforce security.  
-  
--   Controlling optimizations by the just-in-time (JIT) compiler so the code remains easy to debug.  
-  
--   Obtaining information about the caller to a method.  
-  
-## Related Sections  
- For more information, see:  
-  
--   [Creating Custom Attributes (C#)](../../../../csharp/programming-guide/concepts/attributes/creating-custom-attributes.md)  
-  
--   [Accessing Attributes by Using Reflection (C#)](../../../../csharp/programming-guide/concepts/attributes/accessing-attributes-by-using-reflection.md)  
-  
--   [How to: Create a C/C++ Union by Using Attributes (C#)](../../../../csharp/programming-guide/concepts/attributes/how-to-create-a-c-cpp-union-by-using-attributes.md)  
-  
--   [Common Attributes (C#)](../../../../csharp/programming-guide/concepts/attributes/common-attributes.md)  
-  
--   [Caller Information (C#)](../../../../csharp/programming-guide/concepts/caller-information.md)  
-  
-## See Also  
- [C# Programming Guide](../../../../csharp/programming-guide/index.md)  
- [Reflection (C#)](../../../../csharp/programming-guide/concepts/reflection.md)  
- [Attributes](../../../../../docs/standard/attributes/index.md)
+> Regardless of the targets on which `ValidatedContract` is defined to be valid, the `return` target has to be specified, even if `ValidatedContract` were defined to apply only to return values. In other words, the compiler will not use `AttributeUsage` information to resolve ambiguous attribute targets. For more information, see [AttributeUsage (C#)](attributeusage.md).
+
+## Common uses for attributes
+
+The following list includes a few of the common uses of attributes in code:
+
+- Marking methods using the `WebMethod` attribute in Web services to indicate that the method should be callable over the SOAP protocol. For more information, see <xref:System.Web.Services.WebMethodAttribute>.
+- Describing how to marshal method parameters when interoperating with native code. For more information, see <xref:System.Runtime.InteropServices.MarshalAsAttribute>.
+- Describing the COM properties for classes, methods, and interfaces.
+- Calling unmanaged code using the <xref:System.Runtime.InteropServices.DllImportAttribute> class.
+- Describing your assembly in terms of title, version, description, or trademark.
+- Describing which members of a class to serialize for persistence.
+- Describing how to map between class members and XML nodes for XML serialization.
+- Describing the security requirements for methods.
+- Specifying characteristics used to enforce security.
+- Controlling optimizations by the just-in-time (JIT) compiler so the code remains easy to debug.
+- Obtaining information about the caller to a method.
+
+## Related sections
+
+For more information, see:
+
+- [Creating Custom Attributes (C#)](creating-custom-attributes.md)  
```

Comment:
There is the [tutorial on attributes](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/attributes). I think it's worth to be mentioned in this list.

Reply:
I added that link. When I read that tutorial, I didn't see where it should reference the new field: syntax for auto implemented properties.